File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const frequency_map = {}
        for (let num of nums) {
            frequency_map[num] = (frequency_map[num] || 0) + 1
        }

        const arr = Object.entries(frequency_map)
            .map(([num, freq]) => [freq, parseInt(num)])
            .sort((a,b) => b[0]-a[0])
        return arr.slice(0,k).map((pair) => [pair[1]])
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res = []
        for index, num in enumerate(nums): 
            if index > 0 and num == nums[index-1]:
                continue

            l, r = index +1, len(nums)-1

            while l < r:
                total = num + nums[l] + nums[r]

                if total == 0: 
                    res.append([num, nums[l], nums[r]])
                    l+=1
                    r-=1
                    while nums[l] == nums[l-1] and l < r: 
                        l+=1

                elif total < 0: 
                    l+=1
                else: 
                    r-=1

        return res